src/internal/middleware.js

```diff
@@ -37,20 +37,24 @@ export default function sagaMiddlewareFactory(options = {}) {
   function sagaMiddleware({getState, dispatch}) {
     runSagaDynamically = runSaga
     const sagaEmitter = emitter()
+    const sagaDispatch = !sagaMonitor ? dispatch : wrapSagaDispatch(dispatch)
```

Comment:
After some thoughtsI think you're right. I might have some solution to asap which could fix both #598 (ie #602 failing test) as well as #4.
I'm going to merge this and try with the new scheduling idea